src/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -304,7 +312,13 @@ function VisEditor($scope, $route, timefilter, AppState, $location, kbnUrl, $tim
 
       if (id) {
         notify.info('Saved Visualization "' + savedVis.title + '"');
-        if (savedVis.id === $route.current.params.id) {
+        if ($scope.addToDashMode) {
+          const dashboardBaseUrl = chrome.getNavLinkById('kibana:dashboard');
+          // Not using kbnUrl.change here because the dashboardBaseUrl is a full path, not a url suffix.
+          // Rather than guess the right substring, we'll just navigate there directly, just as if the user
+          // clicked the dashboard link in the UI.
+          window.location.href = `${dashboardBaseUrl.lastSubUrl}&${DashboardConsts.ADD_VIS_PARAM}=${savedVis.id}`;
```

Comment:
I think we want to use [`$window`](https://docs.angularjs.org/api/ng/service/$window) instead of `window`, so we can mock it in unit tests.